tools/pythonpkg/src/numpy/array_wrapper.cpp

```diff
@@ -796,14 +866,36 @@ void NumpyResultConversion::Resize(idx_t new_capacity) {
 	capacity = new_capacity;
 }
 
-void NumpyResultConversion::Append(DataChunk &chunk) {
-	if (count + chunk.size() > capacity) {
-		Resize(capacity * 2);
+void NumpyResultConversion::SetCategories() {
+	for (idx_t col_idx = 0; col_idx < owned_data.size(); col_idx++) {
+		auto &type = Type(col_idx);
+		if (type.id() == LogicalTypeId::ENUM) {
+			// It's an ENUM type, in addition to converting the codes we must convert the categories
+			if (categories.find(col_idx) == categories.end()) {
+				auto &categories_list = EnumType::GetValuesInsertOrder(type);
+				auto categories_size = EnumType::GetSize(type);
+				for (idx_t i = 0; i < categories_size; i++) {
+					categories[col_idx].append(py::cast(categories_list.GetValue(i).ToString()));
+				}
+			}
+		}
 	}
+}
+
+void NumpyResultConversion::Append(DataChunk &chunk, idx_t offset) {
+	D_ASSERT(offset < capacity || (offset == 0 && chunk.size() == 0));
 	auto chunk_types = chunk.GetTypes();
 	for (idx_t col_idx = 0; col_idx < owned_data.size(); col_idx++) {
-		owned_data[col_idx].Append(count, chunk.data[col_idx], chunk.size());
+		owned_data[col_idx].Append(offset, chunk.data[col_idx], chunk.size());
+	}
+}
+
+void NumpyResultConversion::Append(DataChunk &chunk) {
+	if (count + chunk.size() > capacity) {
+		py::gil_scoped_acquire gil;
+		Resize(capacity * 2);
```

Comment:
With the batched data collection we should be able to avoid the Resize, since we know the count, right?

Reply:
Yea I don't use this, I moved the appending logic to another function that this one calls after it's done resizing